analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContextBase.cs

```diff
@@ -57,7 +59,7 @@ protected SonarAnalysisContextBase(SonarAnalysisContext analysisContext, TContex
     /// <param name="generatedCodeRecognizer">When set, generated trees are analyzed only when language-specific 'analyzeGeneratedCode' configuration property is also set.</param>
     public bool ShouldAnalyzeTree(SyntaxTree tree, GeneratedCodeRecognizer generatedCodeRecognizer) =>
         (generatedCodeRecognizer is null || SonarLintFile().AnalyzeGeneratedCode || !tree.IsGenerated(generatedCodeRecognizer, Compilation))
```

Comment:
The `SonarLintXmlReader` is retrieved here already, I suggest propagating it to the `IsFileIncluded` method, that way we get rid of a lot of repetitive, redundant calls to the cache to retrieve the reader again and again.